benchmark/BSON/DocumentBench.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Benchmark\BSON;
+
+use MongoDB\Benchmark\BaseBench;
+use MongoDB\BSON\Document;
+use PhpBench\Attributes\BeforeMethods;
+
+use function file_get_contents;
+use function iterator_to_array;
+use function MongoDB\BSON\fromJSON;
+
+#[BeforeMethods('prepareData')]
+final class DocumentBench extends BaseBench
+{
+    private static string $rawBSON;
+    private static Document $document;
+
+    public function prepareData(): void
+    {
+        self::$rawBSON = fromJSON(file_get_contents(self::LARGE_FILE_PATH));
```

Comment:
Was this split out from `getBSONDocument()` to distinguish _reading_ the JSON file from constructing the Document instance?

If neither method is included in the benchmark, I'd have expected you to use `Document::fromJSON()`.

----

Edit: I just noticed the `BeforeMethods('prepareData')` annotation. I guess that explains this.

If you run this method before the benchmark, why do you want `getBSONDocument()` included? I recall you saying that resulted in a slower first run, which balances out in subsequent tests. But in that case do you even want Document constructed counted at all?

Reply:
This was based on a different benchmark which needed extra handling. I simplified the two benchmarks to create the BSON structures directly.